### PR TITLE
Fix flaky test: tui_spec: OptionSet event on terminal-response

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Not sure if that fixes the issue from the failure really, but cannot hurt.

**Comment by blueyed** [comment]
Just failed again on another PR (non-osx):
```
[  ERROR   ] test/functional/terminal/tui_spec.lua @ 1357: TUI background color triggers OptionSet event on terminal-response
test/functional/ui/screen.lua:587: Failed to match any screen lines.
Expected (anywhere): "did OptionSet, yay!"
Actual:
  |                                                  |
  |{4:~                                                 }|
  |{4:~                                                 }|
  |{4:~                                                 }|
  |{5:[No Name]                       0,0-1          All}|
  |:ffff/ffff/ffff{UNEXPECTED foreground = tonumber('0x000051'):^G}{1: }                                |
  |{3:-- TERMINAL --}                                    |
stack traceback:
	test/functional/ui/screen.lua:587: in function '_wait'
	test/functional/ui/screen.lua:370: in function 'expect'
	test/functional/terminal/tui_spec.lua:1375: in function <test/functional/terminal/tui_spec.lua:1357>
```

**Comment by blueyed** [comment]
The test itself appears to have become flaky since recently..

File: test/functional/terminal/tui_spec.lua [hunk 1]
```diff
@@ -1359,7 +1359,7 @@ describe('TUI background color', function()
 
     -- Wait for the child Nvim to register the OptionSet handler.
     feed_data('\027:autocmd OptionSet\n')
-    screen:expect({any='--- Autocommands ---'})
+    screen:expect({any='              echo "did OptionSet, yay!"'})
```
Comment (justinmk) [inline test/functional/terminal/tui_spec.lua]: I don't see how this helps anything. And it makes line 1361 useless...

**Comment by blueyed** [comment]
Failed in first try after 10568.19 ms already.
Rebased to fix conflict, trying the new test.

**Comment by blueyed** [comment]
Well, I think https://github.com/neovim/neovim/pull/11145 would also fix this / help here..
The test sends the bg response manually, although it is not expected to be seen (anymore).

**Comment by blueyed** [comment]
New/rewritten/combined test passed 500 rounds.

@justinmk what do you think?
https://github.com/neovim/neovim/pull/11129/files#diff-8a087fe7dae801e7d38a9d0af6746a8e

File: test/functional/terminal/tui_spec.lua [hunk 2]
```diff
@@ -1345,59 +1345,26 @@ describe("TUI", function()
...
-            {4:~                                                 }|
-            {4:~                                                 }|
-            {5:[No Name]                       0,0-1          All}|
-            %-5s                                             |
-            {3:-- TERMINAL --}                                    |
-          ]], bg)
-        })
-      end)
-    end
+it('TUI background color handles deferred background color', function()
```
Comment (justinmk) [inline test/functional/terminal/tui_spec.lua]: TUI bg color triggers OptionSet event on terminal-response

**Review by justinmk — APPROVED** [review]

**Pull request closed without merging**